### DISCUSSION TITLE
Restoration of `WebviewPanel` html

### DISCUSSION THREAD

**Discussion by senyai** [microsoft/vscode-discussions#207] [Extension Development QnA] [answered]
Hi!

I'm writing .wiki preview extension. How it's working: I subscribe to `workspace.onDidChangeTextDocument` and when this event is fired I render html (using executable file) and post html using `Webview.postMessage`. This works blazingly fast and I think I like it. Now when I move `WebviewPanel` into a background tab and restore it, there's no rendered html. This behavior is well [documented](https://code.visualstudio.com/api/extension-guides/webview#visibility-and-moving).

What I'm looking for is an event that is triggered when `WebviewPanel` is restored or any other thing that will help me update html. I read [markdown-language-features](https://github.com/microsoft/vscode/tree/main/extensions/markdown-language-features) code multiple times, but to little avail. I will appreciate any help.

**Comment by r3m0t** [comment]
https://code.visualstudio.com/api/extension-guides/webview#persistence

**Comment by senyai** [comment]
@r3m0t 
> Use [message passing](https://code.visualstudio.com/api/extension-guides/webview#passing-messages-from-a-webview-to-an-extension) to save off the webview's state and then restore the state when the webview becomes visible again.

That's kind of that I'm asking: 'when the webview becomes visible again?' I thought `deserializeWebviewPanel` will be it, but it's only triggered after vscode is restarted.

**Reply by r3m0t** [reply]
I think you need onDidChangeViewState. And setState in the webview so you can getState and render an initial state,  even if it's outdated.

**Comment by senyai** [comment, marked as answer]
My initial idea that everything works as expected in builtin markdown-language-features extension was wrong. The webview is not updated when it needs to. I need to file a bug report (later). For my case I have two ideas: postMessage from the webview when it loads or use `onDidChangeViewState`. I will be testing my ideas next weeks and post an update here when the extension is out (hopefully).
